src/cli/DB2Session.ts

```diff
@@ -0,0 +1,129 @@
+    };
+
+    /**
+     * Option used in profile creation and commands for port for DB2
+     */
+    public static DB2_OPTION_PORT: ICommandOptionDefinition = {
+        name: "port",
+        aliases: ["P"],
+        description: "The Db2 server port number",
+        type: "number",
+        group: DB2Session.DB2_CONNECTION_OPTION_GROUP
+    };
+
+    /**
+     * Option used in profile creation and commands for username / ID  for DB2
+     */
+    public static DB2_OPTION_USER: ICommandOptionDefinition = {
+        name: "username",
+        aliases: ["user", "u"],
+        description: "The Db2 user ID (may be the same as the TSO login)",
+        type: "string",
+        group: DB2Session.DB2_CONNECTION_OPTION_GROUP
+    };
+
+    /**
+     * Option used in profile creation and commands for password/passphrase for DB2
+     */
+    public static DB2_OPTION_PASS: ICommandOptionDefinition = {
+        name: "password",
+        aliases: ["pass", "pw"],
+        description: "The Db2 password (may be the same as the TSO password)",
+        type: "string",
+        group: DB2Session.DB2_CONNECTION_OPTION_GROUP,
+    };
+
+    /**
+     * Option used in profile creation and commands for rejectUnauthorized setting for connecting to DB2
+     */
+    public static DB2_OPTION_DATABASE: ICommandOptionDefinition = {
+        name: "database",
+        aliases: ["db"],
+        description: "The name of the database",
+        type: "string",
+        group: DB2Session.DB2_CONNECTION_OPTION_GROUP
+    };
+
+    /**
+     * Option used in profile creation and commands for base path setting for connecting to DB2
+     */
+    public static DB2_OPTION_SSL_FILE: ICommandOptionDefinition = {
+        name: "sslFile",
+        aliases: ["ssl"],
+        description: "Path to an SSL Certificate file",
+        type: "string",
+        group: DB2Session.DB2_CONNECTION_OPTION_GROUP
+    };
+
+    /**
+     * Options related to connecting to DB2
+     * These options can be filled in if the user creates a profile
+     */
+    public static DB2_CONNECTION_OPTIONS: ICommandOptionDefinition[] = [
+        DB2Session.DB2_OPTION_HOST,
+        DB2Session.DB2_OPTION_PORT,
+        DB2Session.DB2_OPTION_USER,
+        DB2Session.DB2_OPTION_PASS,
+        DB2Session.DB2_OPTION_DATABASE,
+        DB2Session.DB2_OPTION_SSL_FILE
+    ];
+
+    /**
+     * Create a REST Client Session from given command line arguments or profile.
+     * @static
+     * @param {IProfile} args - The arguments specified by the user
+     * @param {IProfile} profile - The DB2 profile contents
+     * @returns {Session} - A session for usage in the DB2 REST Client
+     */
+    public static createDB2Session(args: ICommandArguments, profile?: IProfile): Session {
+        this.log.info("Creating a DB2 session from cmd arguments or profile");
+        const DB2session = {
+            hostname: args.hostname || (isNullOrUndefined(profile) ? undefined : profile.hostname),
```

Comment:
If there is a profile loaded for a command, after the 'command precedence' changes, the profile fields are automatically filled in to the `args` object. So I think you don't necessarily need these checks for falling back to the profile value. However, it doesn't hurt anything

Reply:
You are right, thanks